media_driver/agnostic/common/codec/hal/codechal_decode_vc1.cpp

```diff
@@ -3222,7 +3222,7 @@ MOS_STATUS CodechalDecodeVc1::SetFrameStates()
     bool pictureIsI = m_mfxInterface->IsVc1IPicture(currPic, m_vc1PicParams->picture_fields.is_first_field, picType) ? true : false;
     bool pictureIsP = m_mfxInterface->IsVc1PPicture(currPic, m_vc1PicParams->picture_fields.is_first_field, picType) ? true : false;
     bool pictureIsB =
-        (m_mfxInterface->IsVc1BPicture(currPic, m_vc1PicParams->picture_fields.is_first_field, picType) |
+        (m_mfxInterface->IsVc1BPicture(currPic, m_vc1PicParams->picture_fields.is_first_field, picType) ||
```

Comment:
Indeed, this looks to be a typo considering these functions are returning bool.

https://github.com/intel/media-driver/blob/ae09cc33e24f4221e083d53a2c5aa035e57cf3f0/media_driver/agnostic/common/hw/vdbox/mhw_vdbox_mfx_interface.h#L780

https://github.com/intel/media-driver/blob/ae09cc33e24f4221e083d53a2c5aa035e57cf3f0/media_driver/agnostic/common/hw/vdbox/mhw_vdbox_mfx_interface.h#L798

I also don't get why to use `()?:` if they are returning bool. I would rewrite the whole this code block as:
```
bool pictureIsI = m_mfxInterface->IsVc1IPicture(currPic, m_vc1PicParams->picture_fields.is_first_field, picType);
bool pictureIsP = m_mfxInterface->IsVc1PPicture(currPic, m_vc1PicParams->picture_fields.is_first_field, picType);
bool pictureIsB =
    m_mfxInterface->IsVc1BPicture(currPic, m_vc1PicParams->picture_fields.is_first_field, picType) ||
    m_mfxInterface->IsVc1BIPicture(currPic, m_vc1PicParams->picture_fields.is_first_field, picType);
```

@mikebenfield : can you, please, help make a change? Also, please, rebase the patch to avoid merge commit in the history.

Reply:
Sure; done.

Reply:
Thank you for rebase. Looks good now. I'll do a rewrite of this code as a follow up after merging in your fix.

Reply:
Oops; sorry, I changed the ternary operator locally as you requested but forgot to commit it to git. Here's a new version with that done.

Reply:
Thank you:).